ouroboros-network-framework/src/Ouroboros/Network/Server2.hs

```diff
@@ -175,6 +177,7 @@ with Arguments {
                           ]
           -- race all `acceptLoops` with `actionThread` and
           -- `inboundGovernorThread`
+          -- TODO is this right?
```

Comment:
This seems to be racing the new action vs the accumulated action, but for that it spawns a new thread that runs the accumulated action. Isn't here an unneeded intermediate thread? Say I have
the list `[a, b]` and the initial element `e`, then this will end up in:
```
(e `race` a) `race` b
```

which will have:
- a racing thread for `b`
- a racing thread for `e race a`
  - a racing thread for `e`
  - a racing thread for `a`

No?

Reply:
We can either try to solve this here or if you prefer leave the `TODO` for later investigation?

 
cc @bolt12

Reply:
Lets leave it, and do it in a different PR

Reply:
Can I do that?

Reply:
In fact this intermediate thread is the one that is unlabelled! This was a nightmare to find.